### ISSUE TITLE
[FEATURE]: Adding keyboard shortcuts for the "Play & Pause" button and "Change Game Speed" button

### ISSUE THREAD

**Issue by Chuenpoly** [COMP3122-2324-Project/Simpsyber#9]
### Description:
![螢幕擷取畫面 2024-02-20 212044](https://github.com/COMP3122-2324-Project/Simpsyber/assets/156781982/27b563b6-f966-40ae-8f5f-90c34e32f078)
Since the timer is still running while players are reassigning the idle staffs, players may frequently stop or slow down the timer first to avoid spending money.

Currently, the game requires players to use mouse to click on the "Play & Pause" button and "Change Game Speed" button to pause the timer or change the game speed. In addition, the operations of the game are mostly controlled by mouse. Players need to frequently move their mouse to click on the buttons. It makes the right hand very busy but the left hand has nothing to do. It is inconvienent and lowers the game experience.

### Suggestion:
To reduce the load of right hand, I suggest adding keyboard shortcuts for the "Play & Pause" button and "Change Game Speed" button. For example, assign spacebar as a shortcut key to play/stop the timer, assign "d" key to speed up the timer and "a" key to slow down the timer. This setting could transfer the load of right hand to left hand and enable players to control the game faster.

**Comment by chunyinng078** [comment]
I agree that using the shortcut key helps reduce the right hand's load. When players/users can use both hands for the game, it could increase the efficiency of controlling the game. But about the shortcut key for controlling the speed, how about using the same button to switch the speed? For example, using the "ctrl" key on the keyboard, click once to switch to a faster speed; when at max speed, the speed goes back to the slowest speed and then loops again. Since when there are more buttons to control on the keyboard, it may reduce the ease of controlling the game with the keyboard, and I am sure more and more shortcut keys will be added to the game.

**Comment by Chuenpoly** [comment]
@chunyinng078 I understand your concern about when there are more buttons to control on the keyboard, it may reduce the ease of controlling the game with keyboard. However, using the same button to switch the speed is not a good idea. For example, in my personal experience, I seldom use speed x2. I only change the speed between normal and speed x1. If there is only one button to switch the speed and it is one-way (i.e. going faster and back to normal speed after max speed), I need to click the button twice back to normal speed when I am using speed x1. Using two buttons, I only need to click "a" button once to back to normal speed in this situation. This is more convenient compared to using one button. Therefore, I think using two buttons to control the speed is better. 

The idea of looping is a good idea because the distance between different speed is only one click. For example, if I am using normal speed, I can click "d" once to speed x1 or click "a" once to speed x2. If I am using speed x1, I can click "a" once to normal speed or click "d" once to speed x2.  If I am using speed x2, I can click "a" once to speed x1 or click "d" once to normal speed.

**Comment by JackYu-QQQ123** [comment]
I agree with your suggestion that adding shortcuts would reduce the burden on the right hand. And it will allow both hands to work together. The time and speed can be adjusted simultaneously by assigning tasks to staff with the mouse.
![Switch](https://github.com/COMP3122-2324-Project/Simpsyber/assets/156782761/ee2c59ff-fd82-49c7-98b8-aa4ee1e506ec)


I also propose integrating shortcuts for quick access to the "Switch to Home" button, enabling us to promptly transition to monitoring our project management. This would expedite access to crucial data such as MTTR, Change Fail Rate, and other pertinent metrics.

**Comment by 22035109d** [comment]
I agree with your point of view. Only when the acceleration key is pressed, will it accelerate; otherwise, time will run normally. When the pause key is pressed, it should pause immediately. The best practice is to, after pressing the acceleration key, wait until someone completes the work, then the system will automatically pause. This can avoid wasting time and money.

**Comment by ChenManHei** [comment]
This is certainly a very great suggestion! Adding keyboard shortcuts for the game function can surely improve the players’ gaming experience.

However, according to the issue mentioned in #4, it seems like that some of the keys in the keyboard like the “ENTER” and “SPACE” key may cause serious malfunctioning of the game. It is very important for the game develop team to solve these types of keyboards causing defects first before adding the keyboard shortcut features in the game, or else it will only bring inconvenience to the players rather than raising their gaming experience.

**Comment by CLee788** [comment]
I agree with adding keyboard shortcuts for the "Play & Pause" and "Change Game Speed" buttons. This suggestion can solve the following problem of user can control to pause the game when entering the test automation page to avoid inconvenience as the user needs to close the test automation page to stop the game. 

https://github.com/COMP3122-2324-Project/Simpsyber/assets/157445841/35f958fc-9291-4204-9634-d5100950d867

**Comment by lordsavor** [comment]
I agree with your idea. As people have mentioned, there are several people that has different ideas on what the default keybind shortcut should be. Additionally, I would suggest to add other shortcuts such as with TAB button tthat allows player to quickly switch to different screens, and a shortcut for test automation tab or leveling up.
